apps/meteor/ee/server/lib/oauth/Manager.ts

```diff
@@ -50,9 +50,12 @@ export class OAuthEEManager {
 				user.roles = [];
 			}
 
+			const savedRoles = Promise.await(Roles.findInIdsOrNames(rolesToSync));
+			const allowedRoles = Promise.await(savedRoles.toArray()).map((role) => role._id);
+
 			Promise.await(
 				syncUserRoles(user._id, rolesFromSSO, {
```

Comment:
```suggestion
			const rolesIdsFromSSO = Promise.await(Roles.findInIdsOrNames(rolesFromSSO).toArray()).map((role) => role._id);
			const allowedRoles = Promise.await(Roles.findInIdsOrNames(rolesToSync).toArray()).map((role) => role._id);

			Promise.await(
				syncUserRoles(user._id, rolesIdsFromSSO, {
```
We should send the roles' ids as params to the `syncUserRoles` function. This works

Reply:
Updated!